mypy/semanal.py

```diff
@@ -3324,8 +3324,16 @@ def name_not_defined(self, name: str, ctx: Context) -> None:
             message += ' {}'.format(extra)
         self.fail(message, ctx)
 
-    def name_already_defined(self, name: str, ctx: Context) -> None:
-        self.fail("Name '{}' already defined".format(name), ctx)
+    def name_already_defined(self, name: str, ctx: Context,
+                             original_ctx: Optional[SymbolTableNode] = None) -> None:
+        if original_ctx:
+            if original_ctx.node and original_ctx.node.get_line() != -1:
+                extra_msg = ' on line {}'.format(original_ctx.node.get_line())
+            else:
+                extra_msg = ' (line unavailable; possibly an import)'
```

Comment:
We could simplify the message a bit. What about just ` (possibly by an import)`. The 'line unavailable' part doesn't provide useful information really.